### PR TITLE
DB Archetypes: README updates for native support

### PR DESCRIPTION
**Pull request by spericas**
Updates to README and related files with a section describing how to run database-se/mp using an external H2 database and GraalVM native.

### CODE REVIEW

File: archetypes/database-mp/src/main/resources/README.md.mustache [hunk 1]
```diff
@@ -38,4 +38,54 @@ curl -s -X GET http://localhost:8080/metrics
...
+to use a TCP connection:
+
+```
+javax.sql.DataSource.test.dataSource.url=jdbc:h2:tcp://localhost:1521/test;IFEXISTS=FALSE
+```
+
+Next, uncomment the following dependencies in your project's pom file:
+
+```
+<dependency>
```
Comment (tomas-langer) [inline archetypes/database-mp/src/main/resources/README.md.mustache:70]: The first dependency is for Oracle JDBC, not needed for h2

File: archetypes/database-mp/src/main/resources/pom.xml.mustache [hunk 2]
```diff
@@ -43,6 +43,14 @@
...
+        <!-- dependency>
+            <groupId>io.helidon.integrations.db</groupId>
+            <artifactId>ojdbc</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.helidon.integrations.db</groupId>
+            <artifactId>h2</artifactId>
+        </dependency -->
         <dependency>
             <groupId>com.h2database</groupId>
```
Comment (tomas-langer) [inline archetypes/database-mp/src/main/resources/pom.xml.mustache:55]: If you use the dependency above, this one is not needed (and you do not need to comment out anything for native image)

File: archetypes/database-mp/src/main/resources/src/main/resources/META-INF/microprofile-config.properties [hunk 3]
```diff
@@ -1,8 +1,8 @@
 
 # Datasource properties
 javax.sql.DataSource.test.dataSourceClassName=org.h2.jdbcx.JdbcDataSource
-#javax.sql.DataSource.test.dataSource.url=jdbc:h2:tcp://localhost:1521/test;IFEXISTS=FALSE
 javax.sql.DataSource.test.dataSource.url=jdbc:h2:mem:test;DB_CLOSE_DELAY=-1
+#javax.sql.DataSource.test.dataSource.url=jdbc:h2:tcp://localhost:1521/test;IFEXISTS=FALSE
```
Comment (tomas-langer) [inline archetypes/database-mp/src/main/resources/src/main/resources/META-INF/microprofile-config.properties:5]: ifexists is not needed here, sufficient on server side

File: archetypes/database-se/src/main/resources/README.md.mustache [hunk 4]
```diff
@@ -38,4 +38,52 @@ curl -s -X GET http://localhost:8080/metrics
...
+The resulting container will listen to port 1521 for network connections.
+Switch the `url` in `application.yaml` to use a TCP connection:
+
+```
+url: jdbc:h2:tcp://localhost:1521/test;DB_CLOSE_DELAY=-1
+```
+
+Next, uncomment the following dependencies in your project's pom file:
+
+```
```
Comment (tomas-langer) [inline archetypes/database-se/src/main/resources/README.md.mustache:68]: Same as above - only one of them is needed (h2), the other one is for Oracle JDBC

File: archetypes/database-mp/src/main/resources/README.md.mustache (hunk 1 above)
Comment (spericas) [reply archetypes/database-mp/src/main/resources/README.md.mustache:70]: Ok, got it.

File: archetypes/database-mp/src/main/resources/src/main/resources/META-INF/microprofile-config.properties (hunk 3 above)
Comment (spericas) [reply archetypes/database-mp/src/main/resources/src/main/resources/META-INF/microprofile-config.properties:5]: That's odd, had issues without it. Will re-try.

File: archetypes/database-mp/src/main/resources/pom.xml.mustache [hunk 5]
```diff
@@ -43,6 +43,10 @@
             <groupId>jakarta.transaction</groupId>
             <artifactId>jakarta.transaction-api</artifactId>
         </dependency>
```
Comment (barchetta) [inline archetypes/database-mp/src/main/resources/pom.xml.mustache:45]: I think you are missing microprofile health and metrics:
```
        <dependency>
            <groupId>io.helidon.microprofile.metrics</groupId>
            <artifactId>helidon-microprofile-metrics</artifactId>
            <scope>runtime</scope>
        </dependency>
        <dependency>
            <groupId>io.helidon.microprofile.health</groupId>
            <artifactId>helidon-microprofile-health</artifactId>
            <scope>runtime</scope>
        </dependency>
```

**Pull request merged**